src/app/clusters/software-diagnostics-server/CodegenIntegration.cpp

```diff
@@ -56,20 +57,42 @@ constexpr bool IsAttributeEnabled(EndpointId endpointId, AttributeId attributeId
 
 } // namespace
 
-void MatterSoftwareDiagnosticsPluginServerInitCallback()
+void emberAfSoftwareDiagnosticsClusterInitCallback(EndpointId endpointId)
 {
-    // NOTE: Code already asserts that only kRootEndpointId is registered.
+    // Restrict cluster to endpoint 0 for dynamic endpoint.
+    if (SoftwareDiagnostics::StaticApplicationConfig::kFixedClusterConfig.size() == 0)
+    {
+        VerifyOrDie(endpointId == kRootEndpointId);
+    }
```

Comment:
```suggestion
    if (endpointId != kRootEndpointId)
    {
        return;
    }
```

Thinking maybe this is more generic. Otherwise we seem to allow odd registrations when the fixed cluster config is as required but we also have dynamic.

Reply:
or `VerifyOrReturn(endpointId == kRootEndpointId);` to be compact

Reply:
Done.

Reply:
Done.